rest/route/host_agent.go

```diff
@@ -1085,6 +1091,25 @@ func sendBackRunningTask(ctx context.Context, env evergreen.Environment, h *host
 		return gimlet.MakeJSONInternalErrorResponder(err)
 	}
 
+	if t.IsStuckTask() {
+		if err := model.ClearAndResetStrandedHostTask(ctx, env.Settings(), h); err != nil {
+			grip.Error(message.WrapError(err, getMessage("ending and resetting system failed task")))
+			return gimlet.MakeJSONInternalErrorResponder(err)
+		}
+		// The agent is expected to run the task once it's been assigned it. If it's requesting the same
+		// task over and over again is a sign that something is wrong.
+		msg := fmt.Sprintf("The agent has re-requested the same task '%d' times.", evergreen.MaxTaskDispatchAttempts)
+		msg += " It's possible that the agent is in a bad state."
+
+		grip.Error(message.Fields{
+			"message":        msg,
+			"task_id":        t.Id,
+			"task_execution": t.Execution,
+			"host_id":        h.Id,
+			"agent_version":  evergreen.AgentVersion,
+		})
```

Comment:
Since it's hit the dispatching limit in this case, is this supposed to return early, or continue with the rest of the logic in the fucntion?